### ISSUE TITLE
Source selection strategy

### ISSUE THREAD

**Issue by schloett** [EEXCESS/recommender#29]
I am a bit confused by the source selection strategy. When the language is set to "english" in the profile, I get results from Mendeley, Deutsche Digitale Bibliothek and ZBW. All results from ZBW are in German. Language of other results is "unknown".
Query:

``` javascript
{
  "contextKeywords":[
    {"weight":1,"text":"Loom"},
    {"weight":1,"text":"Weft"},
    {"weight":1,"text":"Tapestry"}
  ],
  "numResults":50,
  "origin":{
    "module":"FacetScape",
    "clientType":"chrome-extension",
    "clientVersion":"1.2.0",
    "userID":"323F3845-F3C3-42F5-8A72-306493406FFA"
  },
  "languages":[
    {"iso2":"en","languageCompetenceLevel":1}
  ],
  "loggingLevel":0
}
```

Results:
![fs](https://cloud.githubusercontent.com/assets/4618499/11060856/1eda4c62-87a3-11e5-83e5-1ed04e635eb7.PNG)

When I do not send the language in the profile, I get results from more providers and in particular results from Europeana, which are indicated to be English.
Query:

``` javascript
{
  "contextKeywords":[
    {"weight":1,"text":"Loom"},
    {"weight":1,"text":"Weft"},
    {"weight":1,"text":"Tapestry"}
  ],
  "numResults":50,
  "origin":{
    "module":"FacetScape",
    "clientType":"chrome-extension",
    "clientVersion":"1.2.0",
    "userID":"323F3845-F3C3-42F5-8A72-306493406FFA"
  },
  "loggingLevel":0
}
```

Results:
![fs1](https://cloud.githubusercontent.com/assets/4618499/11060898/75c90496-87a3-11e5-8f35-9f0144e3d620.PNG)

Is this intended behavior?

**Comment by hziak** [comment]
yes, the problem is that we do not filter here on the result level, only on what the partner potentially covers.
For example ZBW has German and English content.
So either for German or for English they are selected

Europeana is an exception right now because it was not totally clear which languages Europeana covers.

So in general if a partner covers several languages than he will be selected if one appears in the secure user profile. If the partner has no languages set, he will not be selected at all.
